Header header
int32 keyValue
char keyChar
